luaui/Widgets/cmd_bar_hotkeys.lua

```diff
@@ -25,92 +26,58 @@ local function reloadWidgetsBindings()
 	end
 end
 
-local function makeBindsTable(file)
-	local env = keyConfig.scanToCode[currentLayout]
-	env['table'] = table
-	return VFS.Include(file, env)
-end
 
-local function loadBindingsLegacy(file)
-	for _, v in ipairs(makeBindsTable(file)) do
-		local command = 'bind '..v[1]..' '..v[2]..' '..(v[3] or '')
-		Spring.SendCommands(command)
+local function replaceDefaultWithLegacy(file)
+	if file == 'luaui/configs/hotkeys/default_keys.txt' then
+		return 'luaui/configs/hotkeys/legacy_keys.txt'
 	end
-end
-
--- Map old, deleted lua format presets onto uikeys presets
-local legacyToTxt = {
-	['luaui/configs/bar_hotkeys.lua']          = 'luaui/configs/hotkeys/default_keys.txt',
-	['luaui/configs/bar_hotkeys_60.lua']       = 'luaui/configs/hotkeys/default_keys_60pct.txt',
-	['luaui/configs/bar_hotkeys_grid.lua']     = 'luaui/configs/hotkeys/grid_keys.txt',
-	['luaui/configs/bar_hotkeys_grid_60.lua']  = 'luaui/configs/hotkeys/grid_keys_60pct.txt',
-	['bar_hotkeys_custom.lua']                 = 'uikeys.txt',
-}
-
-
--- TODO: This code is exclusively to convert from the placeholder lua format to the more stable uikeys format
--- TODO: After a period of ~6 months it should be deleted (around January 2024 or later)
-local function replaceLegacyPreset()
-	local keyFile = Spring.GetConfigString("KeybindingFile")
-	-- sometimes this file can be in other directories, we do a string match
-	local isCustom = keyFile and keyFile:match("bar_hotkeys_custom.lua") ~= nil or false
-	if not keyFile or not isCustom then
-		return false
-	end
-
-
-	local newFormat = legacyToTxt[keyFile]
-	if isCustom then -- in case the dir is not root, just pass the filename not the whole path
-		newFormat = legacyToTxt["bar_hotkeys_custom.lua"]
-	end
-	if not newFormat then return false end
-
-	-- in case the user has a uikeys file already, we need to back it up because it could potentially get overwritten by the following steps
-	if VFS.FileExists("uikeys.txt") then
-		Spring.Echo("BAR Hotkeys: Found existing unused uikeys file, creating a backup called uikeys_auto_backup.txt")
-		Spring.SendCommands("keyreload")
-		os.rename("uikeys.txt", "uikeys_auto_backup.txt")
-	end
-
-	-- output the current custom .lua bindings into a uikeys.txt file
-	if isCustom then
-		Spring.Echo("BAR Hotkeys: bar_hotkeys_custom.lua found. This format is deprecated, a " .. newFormat .. " file was written to your bar folder")
-		if VFS.FileExists(keyFile) then
-			Spring.SendCommands("unbindall")
-			loadBindingsLegacy(keyFile)
-		end
-
-		Spring.SendCommands("keysave " .. newFormat)
-	else
-		Spring.SendCommands("keyreload " .. newFormat)
+	if file == 'luaui/configs/hotkeys/default_keys_60pct.txt' then
+		return 'luaui/configs/hotkeys/legacy_keys_60pct.txt'
 	end
+end
 
-	Spring.SetConfigString("KeybindingFile", newFormat)
 
-	return true
+-- if keybinds are missing, load default hotkeys
+local function fallbackToDefault(currentKeys)
+	local default = keyConfig.keybindingLayoutFiles[1]
```

Comment:
I don't like the indexing to set the default, any reason to prefer this instead of an explicit "Grid"?

Reply:
Copy-pasting an above comment:

> My thought with this specifically is that any change to the name of the presets, or any future change to the default preset will result in this breaking if all the appropriate strings are not updated. Using the indexed value means that the first item in that preset list will be treated as default, regardless of name. Granted, the flip side of that is that someone could rearrange that list without realizing this is keyed to it could break things, but I feel like that's a less likely outcome. I'm open to having my mind changed on that though.

Basically if the name of a preset ever gets changed, it will need to be updated in `keyboard-layouts`, `gui_options`, and `cmd_bar_hotkeys`, and that feels like a likely vector for regressions. Having something tied to a variable will be more searchable and findeable, especially since "Grid" is going to turn up like 2000 results in the codebase, where `keybindingLayoutFiles` will be very specific.